components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/listeners/UserManagementV2AuditLogger.java

```diff
@@ -216,10 +218,13 @@ public boolean doPostGetUserClaimValueWithID(String userId, String claim, List<S
             dataObject.put(CLAIM_VALUE_FIELD, new JSONArray(claimValue));
         }
         dataObject.put(PROFILE_FIELD, profileName);
-        AuditLog.AuditLogBuilder auditLogBuilder = new AuditLog.AuditLogBuilder(getInitiatorId(),
-                LoggerUtils.getInitiatorType(getInitiatorId()), userId, LoggerUtils.Target.User.name(),
-                GET_USER_CLAIM_VALUE_ACTION).data(jsonObjectToMap(dataObject));
-        triggerAuditLogEvent(auditLogBuilder);
+        Optional<String> initiatorId = getInitiatorIdForGet();
+        if (initiatorId.isPresent()) {
```

Comment:
Why do we need to check whether initiatorId is present? we need to log the change that is happening here. If the initiatorId is not resent, we need to fix iy

Reply:
Certain cases, it is initiated by System, in those cases, we were earlier logging as SYSTEM. But it was decided that is not necessary.